cranelift-codegen/meta/src/shared/legalize.rs

```diff
@@ -275,6 +275,17 @@ pub(crate) fn define(insts: &InstructionGroup, imm: &Immediates) -> TransformGro
         ],
     );
 
+    narrow.legalize(
+        def!(a = popcnt.I128(x)),
```

Comment:
This is even valid for types other than `I128` when narrowing is needed, for instance for i64 on i686. Can you remove the `.I128` please? (I think it should just work because of our type inference; otherwise some in-flight work is required, and let's forget about this at the moment, see #1119 for context.)

If it does indeed work, can you also add a test for an `popcnt.i64` on i686, please?

Reply:
It does not compile when I remove the `.I128`:

```rust
error[E0308]: mismatched types
    --> /home/kerem/src/cranelift/target/debug/build/cranelift-codegen-f09ff84b74476b73/out/legalizer.rs:2138:60
     |
2138 |                     (Some(a), Some(b)) => a.wider_or_equal(b),
     |                                                            ^ expected struct `ir::types::Type`, found enum `core::option::Option`
     |
     = note: expected type `ir::types::Type`
                found type `core::option::Option<ir::types::Type>`
```

Reply:
Ah, then there are missing bits from this WIP PR mentioned in #1119, we'll worry about this later.